File: Kata 5/Directions Reduction/Directions Reduction.py
def dirReduc(arr):    
    dic={'EAST':'WEST','WEST':'EAST','NORTH':'SOUTH','SOUTH':'NORTH'}
    i=0    
    while i+1<len(arr):
        if arr[i]==dic[arr[i+1]]:
            arr.pop(i)
            arr.pop(i)
            i=0          
        else:
            i+=1            
    return arr
    
    
def dirReduc(arr):
    opposite={'EAST':'WEST','WEST':'EAST','NORTH':'SOUTH','SOUTH':'NORTH'}
    new_plan = []
    for d in arr:
        if new_plan and new_plan[-1] == opposite[d]:
            new_plan.pop()
        else:
            new_plan.append(d)
    return new_plan

File: Kata 5/Directions Reduction/Directions Reduction.cs
public class DirReduction {
  
    public static string[] dirReduc(String[] arr) {
         // "NORTH", "SOUTH", "SOUTH", "EAST", "WEST", "NORTH", "WEST" 
            Dictionary<string, string> dic = new Dictionary<string, string>();
            dic.Add("NORTH", "SOUTH");
            dic.Add("SOUTH", "NORTH");
            dic.Add("WEST", "EAST");
            dic.Add("EAST", "WEST");

            Stack<string> result = new Stack<string>();           

            foreach (string dire in arr)
            {
                if (result.Count > 0 && result.Peek() == dic[dire])                
                    result.Pop();              
                else
                    result.Push(dire);
            }
            
            return result.Reverse().ToArray();
    }
    public static string[] dirReduc2(String[] arr) {
        Dictionary<string, string> dic = new Dictionary<string, string>();
        dic.Add("NORTH", "SOUTH");
        dic.Add("SOUTH", "NORTH");
        dic.Add("WEST", "EAST");
        dic.Add("EAST", "WEST");
        
        List<string> result = new List<string>();

        foreach (string dire in arr)
        {
            if (result.Count > 0 && result[result.Count - 1] == dic[dire])
            {
                 result.RemoveAt(result.Count - 1);
            }
             else
                result.Add(dire);

        }
        return result.ToArray();

     }
}